pkg/controller/autoimport/autoimport_controller.go

```diff
@@ -102,7 +101,7 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
 	}
 
 	reqLogger.V(5).Info("Reconciling auto import secret")
-
+	mcEventRecorder := helpers.NewManagedClusterEventRecorder(r.kubeClient, controllerName, managedCluster)
```

Comment:
TODO: need to shut down the event recorder after ensuring the event is reported. Otherwise, this may result in a memory leak.

Reply:
using the Kubernetes client-go event library can resolve this issue.